src/storage/projectStorage.ts

```diff
@@ -0,0 +1,29 @@
+import { Project } from "../models/projectModels";
+
+export async function getPersistedProjects(): Promise<Project[]> {
+  const result = await chrome.storage.sync.get("projects");
+  if (result.projects !== undefined) {
+    return JSON.parse(result.projects);
+  } else {
+    return [];
+  }
+}
+
+export async function setPersistedProjects(projects: Project[]) {
+  await chrome.storage.sync.set({
+    projects: JSON.stringify(projects),
+  });
+}
+
+export async function showAddProject(): Promise<boolean> {
+  const projects = await getPersistedProjects();
+  const skipAddProjectResult = await chrome.storage.sync.get("skipAddProject");
+  const skipAddProject = JSON.parse(skipAddProjectResult.skipAddProject);
+  return projects.length === 0 && JSON.parse(skipAddProject) === false;
+}
+
+export async function setSkipAddProject(skipAddProject: boolean) {
+  await chrome.storage.sync.set({
+    skipAddProject: JSON.stringify(skipAddProject),
```

Comment:
Wait, is there a reason this needs to be stored in storage? Can’t it be inferred from e.g. the number of projects?

Reply:
I was planning for this to be a boolean flag where if someone doesn't want to track by projects, they could skip the project creation process completely